serverless/src/submitSwodlrOrder/handler.js

```diff
@@ -0,0 +1,190 @@
+          'retrieval_orders.granule_params'
+        )
+        .join('retrieval_collections', { 'retrieval_orders.retrieval_collection_id': 'retrieval_collections.id' })
+        .join('retrievals', { 'retrieval_collections.retrieval_id': 'retrievals.id' })
+        .where({
+          'retrieval_orders.id': id
+        })
+
+      const {
+        access_method: accessMethod,
+        collection_id: collectionConceptId,
+        environment: earthdataEnvironment,
+        granule_params: granuleParams
+      } = retrievalRecord
+
+      const { type, url: swodlrUrl } = accessMethod
+
+      // No need for Accessmethod at this stage of dev.
+      const granuleInfo = await constructOrderPayload({
+        collectionConceptId,
+        granuleParams,
+        accessToken,
+        earthdataEnvironment
+      })
+
+      const { orderItems } = granuleInfo
+
+      await orderItems.forEachAsync(async (granule) => {
+        const { granuleUr, granuleConceptId } = granule
+
+        const splitTitle = granuleUr.split('_')
+        const cycle = parseInt(splitTitle[10], 10)
+        const pass = parseInt(splitTitle[11], 10)
+        const scene = parseInt(splitTitle[12].substring(0, splitTitle[12].length - 1), 10)
+
+        const { json_data: jsonData } = granuleParams
+        const { params, custom_params: customParams } = jsonData
+
+        const {
+          outputGranuleExtentFlag,
+          outputSamplingGridType,
+          rasterResolution
+        } = params
+
+        const {
+          utmZoneAdjust,
+          mgrsBandAdjust
+        } = customParams[granuleConceptId]
+
+        const variables = {
+          cycle,
+          pass,
+          scene,
+          outputGranuleExtentFlag,
+          outputSamplingGridType,
+          rasterResolution,
+          utmZoneAdjust,
+          mgrsBandAdjust
+        }
+
+        const requestId = uuidv4()
+
+        console.log(`Submitting retrieval_order ${id} to swodlr with requestId ${requestId}`)
+
+        // Submit the order
+
+        const response = await axios({
+          url: `${swodlrUrl}/api/graphql`,
+          method: 'post',
+          data: {
+            query: graphQlQuery,
+            variables
+          },
+          headers: {
+            authorization: `Bearer ${accessToken}`,
+            'X-Request-Id': requestId
+          }
+        })
+
+        const { data: responseData, errors } = response
+
+        console.log(errors)
+
+        if (errors) throw new Error(JSON.stringify(errors))
+
+        const { data } = responseData
+        const { generateL2RasterProduct } = data
+
+        // ID Information
+        // product Id is the Id of the product that's being generated
+        // jobId is the Id of the job that is being run to generate the product
+        const { id: productId, status: jobStatus } = generateL2RasterProduct
+        const { id: jobId, state } = jobStatus[0]
+
+        await dbConnection('retrieval_orders').update({
```

Comment:
Does this need to be outside of the foreach? Otherwise we are just overriding this value each granule

Reply:
Good question. Let me think about it.